AddNewRecipes/Program.cs

```diff
@@ -45,39 +46,39 @@ public static async Task<int> Main(string[] args)
         public static void RunPatch(IPatcherState<ISkyrimMod, ISkyrimModGetter> state)
         {
             List<IngrCombination> combinations = new List<IngrCombination>();
-            IEnumerable<IIngredientGetter> ingredients = state.LoadOrder.PriorityOrder.OnlyEnabled().Ingredient().WinningOverrides();
-            ingredients = from ingrs in ingredients where !SkipPlugins.Contains(ingrs.FormKey.ModKey.Name.ToLower()) select ingrs;
-            ingredients = from ingrs in ingredients where (!SkipIngrs.Intersect(ingrs.Name?.ToString()?.Split()!).Any() || SkipIngrs.Contains(ingrs.Name?.ToString())) select ingrs;
-            ingredients = from ingrs in ingredients where !String.IsNullOrEmpty(ingrs.Name?.String) select ingrs;
-            IEnumerator<IIngredientGetter> enumerator = ingredients.GetEnumerator();
+            var ingredients = state.LoadOrder.PriorityOrder.OnlyEnabled().Ingredient().WinningOverrides()
+                .Where(x => !SkipPlugins.Contains(x.FormKey.ModKey.Name.ToLower()))
+                .Where(x => (!SkipIngrs.Intersect(x.Name?.ToString()?.Split()!).Any() || SkipIngrs.Contains(x.Name?.ToString())))
+                .Where(x => !String.IsNullOrEmpty(x.Name?.String))
+                .ToList();
             int i = 0;
-            int percent = (int)(ingredients.Count() * outputPercentage);
-            while (enumerator.MoveNext())
+            int percent = (int)(ingredients.Count * outputPercentage);
+            foreach (var target in ingredients)
             {
                 sw.Start();
                 if (i % percent == 0)
                 {
-                    Console.WriteLine(i + " out of " + ingredients.Count() + " ingredients processed.");
+                    Console.WriteLine(i + " out of " + ingredients.Count + " ingredients processed.");
                 }
-                List<IIngredientGetter> remainingingr = ingredients.Skip(i).ToList();
-                IIngredientGetter[] potionRecipeList = getIngredientsMatchingOneIngredient(enumerator.Current, remainingingr);
-                if (String.IsNullOrEmpty(enumerator.Current.Name?.String))
+                List<IIngredientGetter> remainingingr = ingredients.GetRange(i, ingredients.Count - i);
```

Comment:
This is another example of a misuse of Linq, where it CAN be used to do lots of things when dealing with Enumerables, but often the containers have more optimal ways of doing similar things.  For example, here you want to grab the last X amount of things.

The linq setup you have here fulfills that, but how does it fulfill Skip()?   Well, it literally WALKS/Loops over every value starting from the first value, to the 2nd, to the 3rd, etc, until it gets to `i`, and then finally starts grabbing things to put in the list.

But what is a list?  It's an array, and arrays know exactly how many items they have, and where they are.  Why do you need to loop/walk past the first i items?  Just go straight to i and start looping from there.   That's instant, compared to the Skip() Linq operator.